actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -59,17 +59,23 @@ def collection_by_cache_keys(view, template)
         seed = callable_cache_key? ? @options[:cached] : ->(i) { i }
 
         digest_path = view.digest_path_from_template(template)
+        view_parts = view.combined_fragment_cache_key(nil)
+        base = view_parts.shift
 
         @collection.each_with_object([{}, []]) do |item, (hash, ordered_keys)|
-          key = expanded_cache_key(seed.call(item), view, template, digest_path)
+          key = ActiveSupport::Cache::Key.new(base)
+          view_parts.each do |part|
+            key << part
+          end
```

Comment:
Could add a `concat` method to `Key` that delegates to `concat` on `@key_parts` and do `key.concat view_parts` should be clearer and quicker still.